tests/test_external_search.py

```diff
@@ -1221,14 +1306,35 @@ def query(self):
         # The return value is a new MockSearch object based on the one
         # that was passed in.
         assert isinstance(built, MockSearch)
-        eq_(search, built.parent)
+        eq_(search, built.parent.parent)
```

Comment:
There's now an extra step -- a nested filter that's always applied -- so the mock object ends up creating a child _and_ a grandchild, which is what we end up with as `built`.